### PR TITLE
{math}[fosscuda/2020a,fosscuda/2020b] magma v2.5.4

### PR DESCRIPTION
**Pull request by Micket**
(created using `eb --new-pr`)

### CODE REVIEW

**Comment by Micket** [comment]
@Flamefire you might be interested in this one. 

I started working on an easyblock but, then I noticed we had convenient variables so i didn't have to bother. Looks to me that this can be dropped down to `gcccuda` level.
I also added Turing here (7.5), which is missing from previous magma.

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/7233ec8999b73f892441662ebe051bf7 for a full test report.

**Comment by Flamefire** [comment]
Magma seems to require OpenBLAS and hence must be a bit higher. The failure is due to e.g https://icl.cs.utk.edu/magma/forum/viewtopic.php?t=66&start=10 --> Missing symbols

**Comment by Micket** [comment]
Dangit

**Comment by Flamefire** [comment]
BTW: The cmake output is quite informative. Check the log for the `cmake ` invocation:

```
-- Found CUDA: /sw/installed/CUDAcore/11.0.2 (found version "11.0") 
-- Found CUDA 11.0
--     CUDA_INCLUDE_DIRS:   /sw/installed/CUDAcore/11.0.2/include
--     CUDA_CUDART_LIBRARY: /sw/installed/CUDAcore/11.0.2/lib64/libcudart.so
--     compile for CUDA arch 6.1 (Pascal)
--     compile for CUDA arch 7.0 (Volta)
-- Define -DHAVE_CUBLAS -DMIN_CUDA_ARCH=610
-- Searching for BLAS and LAPACK. To override, set LAPACK_LIBRARIES using ccmake.
-- Looking for Fortran sgemm
-- Looking for Fortran sgemm - not found
-- Could NOT find BLAS (missing: BLAS_LIBRARIES) 
-- LAPACK requires BLAS
-- A library with LAPACK API not found. Please specify library location.
```

**Comment by Micket** [comment]
Test report by @Micket
**FAILED**
Build succeeded for 0 out of 2 (2 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/daa59a3f89d2ef8732f3d602fdc740b3 for a full test report.

**Comment by Flamefire** [comment]
For this a patch to magma is required. Otherwise it will silently drop sm specifications. See https://bitbucket.org/icl/magma/issues/32/list-of-whitelisted-sm_-values-for-cuda-is and https://github.com/easybuilders/easybuild-easyconfigs/pull/11861

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0306u19a - Linux RHEL 8.2, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/8e45356629cb77bd7f6289bb5e9355b1 for a full test report.

**Comment by Micket** [comment]
@branfosj Please hold off on this until we've looked into patching magma (just the cmakelists file that unnecessarily limits the targets i think)

**Comment by Micket** [comment]
Test report by @Micket
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
alvis-c1 - Linux centos linux 7.9.2009, x86_64, Intel Xeon Processor (Skylake), Python 3.6.8
See https://gist.github.com/21913d2fa3f351fc1cacec360f8f98b8 for a full test report.

**Review by branfosj — APPROVED** [review]
lgtm

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
bear-pg0306u03a.bear.cluster - Linux RHEL 8.2, POWER, 8335-GTX (power9le), Python 3.6.8
See https://gist.github.com/e2d0c5edeffe609e465679e4d180d627 for a full test report.

File: easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+easyblock = "CMakeMake"
```
Comment (Flamefire) [inline easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb:1]: Nit: Why the change to double quotes here after everything else is already single quote?

**Comment by branfosj** [comment]
Test report by @branfosj
**SUCCESS**
Build succeeded for 2 out of 2 (2 easyconfigs in total)
bear-pg0212u17a.bear.cluster - Linux centos linux 8.2.2004, x86_64, Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz (broadwell), Python 3.6.8
See https://gist.github.com/2792abd67dc72f70990897a52f857e05 for a full test report.

File: easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb (hunk 1 above)
Comment (Micket) [reply easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb:1]: I just rebased them on your updated 2019b version and it just followed from that. I have no strong opinions. I can change it if other care?

Comment (branfosj) [reply easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb:1]: Double quotes for the easyblock is consistent with everyother magma easyconfig:
```
$ grep easyblock *
magma-2.3.0-foss-2018a-CUDA-9.1.85.eb:easyblock = "ConfigureMake"
magma-2.3.0-fosscuda-2017b.eb:easyblock = "ConfigureMake"
magma-2.3.0-intelcuda-2017b.eb:easyblock = "ConfigureMake"
magma-2.4.0-fosscuda-2018b.eb:easyblock = "ConfigureMake"
magma-2.5.0-fosscuda-2018b.eb:easyblock = "ConfigureMake"
magma-2.5.1-fosscuda-2019a.eb:easyblock = "CMakeMake"
magma-2.5.1-fosscuda-2019b.eb:easyblock = "CMakeMake"
magma-2.5.4-fosscuda-2019a.eb:easyblock = "CMakeMake"
magma-2.5.4-fosscuda-2019b.eb:easyblock = "CMakeMake"
```

I do not mind either way.

Comment (Flamefire) [reply easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb:1]: Then I must have overlooked that. If you don't mind I'd say change it so it is consistent  and new ECs based on this (this beeing the most recent ones) will have that too. No retest required anyway

**Comment by branfosj** [comment]
Going in, thanks @Micket!

File: easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb (hunk 1 above)
Comment (boegel) [reply easybuild/easyconfigs/m/magma/magma-2.5.4-fosscuda-2020a.eb:1]: FWIW: I'm in favor of single quotes for string values that don't have spaces, but unless we make the style check test trip over that, it's pointless to discuss aspects like this.

Sticking to double quotes makes sense, it allows easier review with [`--review-pr`](https://docs.easybuild.io/en/latest/Integration_with_GitHub.html#reviewing-easyconfig-pull-requests-review-pr)
